manifests/user/crontab.pp

```diff
@@ -0,0 +1,93 @@
+# == Class: cron::user::crontab
+#
+#  About:  This class is to manage user crontabs.
+#
+#  Purpose:  Often application teams or application users
+#            have crontabs that must execute as the application user.
+#            For example:  DBAs may want to execute crontab entries
+#            as the user oracle, mysql, etc.  Same goes with other
+#            commercial and proprietary applications operating in a
+#            Least Required Permissions model.
+#
+#  Usage:
+#      $user_crontabs => {
+#          'user1' => {'vars' => [ 'SHELL': '/bin/bash', 'MYECHO': '$(which echo)' ], 'entries' => [ '# Echo Hello World': '* 1 * * * $MYECHO "Hello World!"' ]},
+#          'user2' => {'vars' => [ 'SHELL': '/bin/bash', 'MYECHO': '$(which echo)' ], 'entries' => [ '# Echo Hello World': '* 3 * * * $MYECHO "Hello user2!"' ]}
+#      }
+#      create_resources(cron::user::crontab, $user_crontabs)
+#
+#
+#  Hiera data structure to be used by the parent class cron:
+#      cron::user_crontabs:
+#        'user1':
+#          vars:
+#            'SHELL': '/bin/bash'
+#            'MYECHO': '$(which echo)'
+#          entries:
+#            '# Echo Hello World': '* 1 * * * $MYECHO "Hello World!" 2>&1'
+#        'user2':
+#          vars:
+#            'SHELL': '/bin/bash'
+#            'MYECHO': '$(which echo)'
+#          entries:
+#            '# Echo Hello World': '* 3 * * * $MYECHO "Hello user2!" 2>&1'
+#
+#
+#
+define cron::user::crontab (
+  $ensure  = file,
+  $owner   = undef,
+  $group   = undef,
+  $mode    = '0600',
+  $path    = $cron::user_crontab_path,
+  $content = undef,
+  $vars    = undef,
+  $entries = undef,
+){
+
+  include ::cron
+
+  if $owner == undef {
+    $myowner = $name
+  }
+  else {
+    $myowner = $owner
+  }
+
+  if $group == undef {
+    $mygroup = $name
+  }
+  else {
+    $mygroup = $group
+  }
+
+  if $entries != undef {
+    validate_hash($entries)
+    $crontab_tasks = $entries
```

Comment:
I certainly agree with that.  I originally had completely separate templates and developed separately to not interfere with anything working.  Then I started using the existing crontab.erb which resulted in the variable rename seen there.  I did leave it to discuss potential confusion with parent class variables named the same since $entries is a parameter within the user_crontabs hash.  

I can certainly change it without any issues.

Reply:
I have no hard opinion on that ;) Just an idea that came to my mind.